lib/modules/navigationMiddleware.js

```diff
@@ -57,10 +61,20 @@ const findAndCallHandler = (store, routes, shouldSetPage, data) => {
       getState
     );
 
-    if (h[method] === undefined) {
-      throw new Error(`No method found for ${method.toUpperCase()} ${pathName}`);
+    let handlerMethod = method;
+    // HEAD requests are supposed to have the exact same headers and redirects
+    // as a GET request, but they must not send a response body.
+    // To support HEAD requests, we can check if the handler
+    // has a specific HEAD function, otherwise we just use its GET function
+    if (handlerMethod === METHODS.HEAD && !h[METHODS.HEAD] && !!h[METHODS.GET]) {
+      handlerMethod = METHODS.GET;
     }
-    return h[method].bind(h);
+
+    if (h[handlerMethod] === undefined) {
```

Comment:
ah. well, if you don't think it's crazy, we could just make that a falsey check to keep things simple.